### ISSUE TITLE
Focus error

### ISSUE THREAD

**Issue by michele-pauluzzi** [bostrowski/Pretius-APEX-Enhanced-Lov-Item#17]
If you click on the item more than one time you create another prompt underneat the first one, if you then type something and click away from it, you will get an error.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/54415530/63540308-2c082c00-c51c-11e9-939e-262aab7dbcc6.gif)

Temporary Fix:
add  this.prompt.input.focus(); to line 4348
change this 
```
_promptSearchInputFocusHandler: function( pEvent ){
    apex.debug.log(this.logPrefix, '_promptSearchInputFocusHandler');

    if ( this.mask.state.disabled ) {
      apex.debug.log(this.logPrefix, '_promptSearchInputFocusHandler', 'APEX item is disabled, do nothing.');
      return void(0);
    }

    if ( !this.pluginSettings.isAutocompleteAvailable ) {
      apex.debug.log(this.logPrefix, '_promptSearchInputFocusHandler', 'Autocomplete is not available');
      return void(0);
    }

    if ( this.prompt.container == undefined ) {

      this._promptCreateNew();
      this._promptAddListeners();

      this._promptShow();

      if (
        this.pluginSettings.prompt.isMinimalInputLength
        && this.pluginSettings.prompt.minimalInputLength == 0
      ){
        this._promptDrawMinimalInputLength( this.C_PROMPT_TEXT_MINIMAL_INPUT_LENGTH_0 );
      }
      else if ( 
        this.pluginSettings.prompt.isMinimalInputLength 
        && this._promptIsMinimalInputLengthMatched()
      ) {
        this._promptDrawMinimalInputLength( this.C_PROMPT_TEXT_MINIMAL_INPUT_LENGTH_X );
      }
      else {
        this._promptAjaxPerform();
      }

      this.prompt.input.focus();
    }    
  },
```

to this

```
  _promptSearchInputFocusHandler: function( pEvent ){
    apex.debug.log(this.logPrefix, '_promptSearchInputFocusHandler');

    if ( this.mask.state.disabled ) {
      apex.debug.log(this.logPrefix, '_promptSearchInputFocusHandler', 'APEX item is disabled, do nothing.');
      return void(0);
    }

    if ( !this.pluginSettings.isAutocompleteAvailable ) {
      apex.debug.log(this.logPrefix, '_promptSearchInputFocusHandler', 'Autocomplete is not available');
      return void(0);
    }

    if ( this.prompt.container == undefined ) {

      this._promptCreateNew();
      this._promptAddListeners();

      this._promptShow();

      if (
        this.pluginSettings.prompt.isMinimalInputLength
        && this.pluginSettings.prompt.minimalInputLength == 0
      ){
        this._promptDrawMinimalInputLength( this.C_PROMPT_TEXT_MINIMAL_INPUT_LENGTH_0 );
      }
      else if ( 
        this.pluginSettings.prompt.isMinimalInputLength 
        && this._promptIsMinimalInputLengthMatched()
      ) {
        this._promptDrawMinimalInputLength( this.C_PROMPT_TEXT_MINIMAL_INPUT_LENGTH_X );
      }
      else {
        this._promptAjaxPerform();
      }

      this.prompt.input.focus();
    }   
      this.prompt.input.focus();
  },
```
Same problem when you select an item from the list, cancel the tag, type something and click away.
![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/54415530/63541975-822a9e80-c51f-11e9-905a-7301ba5a2036.gif)

Same problem when you click on the first item lov, click on another item lov, type something and the focus is still on the first one.

![ezgif com-video-to-gif(2)](https://user-images.githubusercontent.com/54415530/63542242-39bfb080-c520-11e9-86ba-88182e0387f7.gif)

With the fix you then have others problems with the prompt remaining active, so i added this line of code
   $('div.t-Body-contentInner').trigger('click');
on line 3055

```

_promptCreateNew: function(){
    apex.debug.log(this.logPrefix, '_promptCreateNew');

    var 
      attributes  = this.element.get(0).attributes,
      classes     = attributes.class != undefined ? attributes.class.value.split(' ') : [],
      fixBorder   = $('<div class="fixBorder"></div>'),
      container   = $('<div></div>'),
      header      = $('<div></div>'),
      menu        = $('<div class="menu"></div>'),
      menuExpand      = $(''+
        '<a href="javascript: void(0)" class="t-Button t-Button--tiny t-Button--link t-Button--pill view" title="'+this.C_PROMPT_MENU_EXPAND_TITLE+'">'+
        '  <span aria-hidden="true" class="fa fa-compress collapsed"></span>'+
        '  <span aria-hidden="true" class="fa fa-expand expanded"></span>'+
        '</a>'),
      menuSort    = $(''+
        '<a href="javascript: void(0)" class="t-Button t-Button--tiny t-Button--link t-Button--pill sort" title="'+this.C_PROMPT_MENU_SORT_TITLE+'">'+
        '  <span aria-hidden="true" class="a-Icon icon-rpt-sort-asc asc"></span>'+
        '  <span aria-hidden="true" class="a-Icon icon-rpt-sort-desc desc"></span>'+
        '</a>'),
      menuClear    = $(''+
        '<a href="javascript: void(0)" class="t-Button t-Button--tiny t-Button--link t-Button--pill" title="'+this.C_PROMPT_MENU_CLEAR_TITLE+'">'+
        '  <span aria-hidden="true" class="fa fa-trash-o clear"></span>'+
        '</a>'),

      inputSearch = $('<input type="text" tabindex="0">'),
      icon        = $('<span class="fa fa-search"></span>'),
      body        = $('<div><!--default body--></div>'),
      offset      = this.mask.container.offset();

    if ( this.mask.container.is('.expanded') ) {
      menuExpand.addClass('expanded')
    }

    if ( this.mask.state.sorted == undefined ) {
      menuSort.addClass('asc');
    }
    else if ( this.mask.state.sorted == 'asc' ) {
      menuSort.addClass('asc'); 
    }
    else {
      menuSort.addClass('desc');
    }

    menu
      .append(menuExpand)
      .append(menuSort)
      .append(menuClear);

    container
      .addClass('pretius--enhancedLovItem prompt')
      .css({
        "minWidth"   : this.mask.container.outerWidth(),
        "borderColor": this.inputCss.borderColor,
        "fontSize"   : this.inputCss.fontSize,
        "lineHeight" : this.inputCss.lineHeight
      })
      .append(fixBorder.css("borderColor", this.inputCss.borderColor))
    ;

    apex.debug.log(this.logPrefix, '_promptCreateNew', 'offset', offset);


    body
      .addClass('body')
      .css({
        'maxHeight': this.pluginSettings.prompt.maxHeight
      });


    if ( attributes.maxlength != undefined ) {
      inputSearch.attr('maxlength', attributes.maxlength.value);  
    }
    
    if ( attributes.placeholder != undefined ) {
      inputSearch.attr('placeholder', attributes.placeholder.value);  
    }

    for (var i in classes) {
      inputSearch.addClass( classes[i]+'-input' );  
    }

    inputSearch.attr('autocomplete', 'off');
    

    header
      .addClass('header')
      .append( inputSearch )
      .append( icon );


    if ( this.pluginSettings.isMultipleSelection ) {
      container.append(menu);

    }
    container
      .append(header)
      .append(body);
    
    $.extend(this.prompt, {
      "container" : container,
      "header"    : header,
      "body"      : body,
      "input"     : inputSearch,
      "icon"      : icon,
      "fixBorder" : fixBorder,
      "menu"      : {
        "container": menu,
        "expand"   : menuExpand,
        "sort"     : menuSort,
        "clear"    : menuClear
      }
    });
  },
```

to this
```

 _promptCreateNew: function(){
    apex.debug.log(this.logPrefix, '_promptCreateNew');
    $('div.t-Body-contentInner').trigger('click');

    var 
      attributes  = this.element.get(0).attributes,
      classes     = attributes.class != undefined ? attributes.class.value.split(' ') : [],
      fixBorder   = $('<div class="fixBorder"></div>'),
      container   = $('<div></div>'),
      header      = $('<div></div>'),
      menu        = $('<div class="menu"></div>'),
      menuExpand      = $(''+
        '<a href="javascript: void(0)" class="t-Button t-Button--tiny t-Button--link t-Button--pill view" title="'+this.C_PROMPT_MENU_EXPAND_TITLE+'">'+
        '  <span aria-hidden="true" class="fa fa-compress collapsed"></span>'+
        '  <span aria-hidden="true" class="fa fa-expand expanded"></span>'+
        '</a>'),
      menuSort    = $(''+
        '<a href="javascript: void(0)" class="t-Button t-Button--tiny t-Button--link t-Button--pill sort" title="'+this.C_PROMPT_MENU_SORT_TITLE+'">'+
        '  <span aria-hidden="true" class="a-Icon icon-rpt-sort-asc asc"></span>'+
        '  <span aria-hidden="true" class="a-Icon icon-rpt-sort-desc desc"></span>'+
        '</a>'),
      menuClear    = $(''+
        '<a href="javascript: void(0)" class="t-Button t-Button--tiny t-Button--link t-Button--pill" title="'+this.C_PROMPT_MENU_CLEAR_TITLE+'">'+
        '  <span aria-hidden="true" class="fa fa-trash-o clear"></span>'+
        '</a>'),

      inputSearch = $('<input type="text" tabindex="0">'),
      icon        = $('<span class="fa fa-search"></span>'),
      body        = $('<div><!--default body--></div>'),
      offset      = this.mask.container.offset();

    if ( this.mask.container.is('.expanded') ) {
      menuExpand.addClass('expanded')
    }

    if ( this.mask.state.sorted == undefined ) {
      menuSort.addClass('asc');
    }
    else if ( this.mask.state.sorted == 'asc' ) {
      menuSort.addClass('asc'); 
    }
    else {
      menuSort.addClass('desc');
    }

    menu
      .append(menuExpand)
      .append(menuSort)
      .append(menuClear);

    container
      .addClass('pretius--enhancedLovItem prompt')
      .css({
        "minWidth"   : this.mask.container.outerWidth(),
        "borderColor": this.inputCss.borderColor,
        "fontSize"   : this.inputCss.fontSize,
        "lineHeight" : this.inputCss.lineHeight
      })
      .append(fixBorder.css("borderColor", this.inputCss.borderColor))
    ;
	

    apex.debug.log(this.logPrefix, '_promptCreateNew', 'offset', offset);


    body
      .addClass('body')
      .css({
        'maxHeight': this.pluginSettings.prompt.maxHeight
      });


    if ( attributes.maxlength != undefined ) {
      inputSearch.attr('maxlength', attributes.maxlength.value);  
    }
    
    if ( attributes.placeholder != undefined ) {
      inputSearch.attr('placeholder', attributes.placeholder.value);  
    }

    for (var i in classes) {
      inputSearch.addClass( classes[i]+'-input' );  
    }

    inputSearch.attr('autocomplete', 'off');
    

    header
      .addClass('header')
      .append( inputSearch )
      .append( icon );


    if ( this.pluginSettings.isMultipleSelection ) {
      container.append(menu);

    }
    container
      .append(header)
      .append(body);
    
    $.extend(this.prompt, {
      "container" : container,
      "header"    : header,
      "body"      : body,
      "input"     : inputSearch,
      "icon"      : icon,
      "fixBorder" : fixBorder,
      "menu"      : {
        "container": menu,
        "expand"   : menuExpand,
        "sort"     : menuSort,
        "clear"    : menuClear
      }
    });
	
  },
```
I dont know javascript, but at the moment this fix the various problems.

**Comment by bostrowski** [comment]
@michele-pauluzzi 
thank you for your detailed feedback. Will work on it!

**Comment by michele-pauluzzi** [comment]
Thank you for this plugin, it's very useful.

**Comment by michele-pauluzzi** [comment]
Waiting for the new version, keep up the good work.

**Comment by tiagosteil** [comment]
Hi @bostrowski , tanks for the quick reply! 

I already follow you on Twitter, I will wait for the new version, thank you very much for your attention.

**Comment by michele-pauluzzi** [comment]
With the new version this is fixed.

**Comment by bostrowski** [comment]
@michele-pauluzzi 
you mean pre release for v1.0.5 or release I'm still working on v1.1.0?

**Comment by michele-pauluzzi** [comment]
i have tried the version published yesterday v1.1.0

All the new error and change is for that version.

If is not finished i will test again at that time.

**Comment by bostrowski** [comment]
I'm preparing the release. Download it next week after I announce it at twitter :-)
